l3kernel/l3keys.dtx

```diff
@@ -1530,9 +1530,9 @@
   { \exp_args:No \@@_define:nnn \l_@@_module_str }
 \cs_new_protected:Npn \@@_define:nnn #1#2#3
   {
-    \cs_set_nopar:Npx \l_@@_module_str { \@@_trim_spaces:n {#2} }
+    \str_set:Nx \l_@@_module_str { \@@_trim_spaces:n {#2} }
     \keyval_parse:NNn \@@_define:n \@@_define:nn {#3}
-    \cs_set_nopar:Npn \l_@@_module_str {#1}
+    \str_set:Nn \l_@@_module_str {#1}
```

Comment:
I think you missed some related changes for `@@_set:nnn`.  You also missed some `\use:e`→`\use:x` and perhaps some more.  With `git diff HEAD~2` you can see the total changes from 2 commits prior (obviously works for more than 2) and thus more easily see what may need to be reconsidered after my review.  Let me know if you want me to be more thorough pointing out places.

Reply:
I did this commit on a bus ride, so it is very likely I've missed things :) Would using `\__kernel_tl_set:Nx` be fine to set a `str` if one knows that the argument is already a string?

Reply:
I think it's ok (and I'm ready to defend this to Frank et al), since we document that strings are particular kinds of tl.